PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/mysql/creation/PasswordUtils.java

```diff
@@ -87,17 +88,14 @@ private static AzureValidationInfo validatePassword(String password, String user
                     "English uppercase letters, English lowercase letters, numbers (0-9), and non-alphanumeric characters (!, $, #, %, etc.).")
                     .type(AzureValidationInfo.Type.ERROR).build();
         }
-        // validate part of admin username
-        final int adminUsernameLength = StringUtils.length(username);
-        if (adminUsernameLength >= LONGEST_COM_SUB_SEQUENCE_BETWEEN_NAME_AND_PASSWORD) {
-            for (int index = 0; index <= adminUsernameLength - LONGEST_COM_SUB_SEQUENCE_BETWEEN_NAME_AND_PASSWORD; index++) {
-                if (StringUtils.contains(password, StringUtils.substring(username, index, index + LONGEST_COM_SUB_SEQUENCE_BETWEEN_NAME_AND_PASSWORD))) {
-                    final AzureValidationInfo.AzureValidationInfoBuilder builder = AzureValidationInfo.builder();
-                    return builder.input(input).message("Your password cannot contain all or part of the login name." +
-                            " Part of a login name is defined as three or more consecutive alphanumeric characters.")
-                            .type(AzureValidationInfo.Type.ERROR).build();
-                }
-            }
+        // validate longest common subsequence between username and password.
+        LongestCommonSubsequence algorithm = new LongestCommonSubsequence();
+        int longestCommonSubsequenceLength = algorithm.apply(username, password);
+        if (longestCommonSubsequenceLength >= LONGEST_COMMON_SUBSEQUENCE_BETWEEN_NAME_AND_PASSWORD) {
+            final AzureValidationInfo.AzureValidationInfoBuilder builder = AzureValidationInfo.builder();
+            return builder.input(input).message("Your password cannot contain all or part of the login name." +
+                    " Part of a login name is defined as three or more consecutive alphanumeric characters.")
```

Comment:
use LONGEST_COMMON_SUBSEQUENCE_BETWEEN_NAME_AND_PASSWORD here?

Reply:
as discussed IM, keep "three" rather than "3" in this message. keeping it same with portal currently